### PR TITLE
feat(frontend): validateConvertAmount util

### PR DESCRIPTION
**Pull request by DenysKarmazynDFINITY**
# Motivation

The util that will be used in `ConvertAmountSource` as `customValidate` function for the input. I extracted it from the component for the unit testing purposes.

### CODE REVIEW

File: src/frontend/src/lib/utils/convert.utils.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	balance,
+	totalFee
+}: {
+	userAmount: BigNumber;
+	decimals: number;
+	balance?: BigNumber;
+	totalFee?: bigint;
+}): ConvertAmountErrorType => {
+	if (isNullish(totalFee)) {
+		return;
```
Comment (AntonioVentilii) [inline src/frontend/src/lib/utils/convert.utils.ts:20]: I saw that in the code we do the same elsewhere, but I don't remember why? is it possible that the validation comes in another component when the fee is nullish?

File: src/frontend/src/lib/utils/convert.utils.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+				}),
+				decimals
+			)
+		: ZERO;
+
+	if (userAmount.gt(parsedSendBalance)) {
+		return 'insufficient-funds';
+	}
+
+	if (nonNullish(totalFee) && userAmount.add(totalFee).gt(parsedSendBalance)) {
```
Comment (AntonioVentilii) [inline src/frontend/src/lib/utils/convert.utils.ts:38]: theoretically here the `nonNullish` function is not needed, right? you already checked above

**Review by AntonioVentilii — COMMENTED** [review]
In component `SendAmount` we have a similar logic: is there any way that the two functions can be merged? Meaning, can we use your function there too?

File: src/frontend/src/lib/utils/convert.utils.ts (hunk 2 above)
Comment (DenysKarmazynDFINITY) [reply src/frontend/src/lib/utils/convert.utils.ts:38]: ah, good catch, removing it!

**Comment by DenysKarmazynDFINITY** [comment]
@AntonioVentilii they are a few differences between them (mostly, in return types - one uses `strings`, and another one `Error`), but my idea was that they indeed can be standardised and reused in the future. I added a TODO comment for now.

File: src/frontend/src/lib/utils/convert.utils.ts (hunk 1 above)
Comment (DenysKarmazynDFINITY) [reply src/frontend/src/lib/utils/convert.utils.ts:20]: That's a good point. After thinking about it, I think it makes sense to validate `insufficient-funds` case even if `totalFee` is nullish. Therefore, I removed this check and adjusted the tests.

File: src/frontend/src/lib/utils/convert.utils.ts (hunk 2 above)
Comment (DenysKarmazynDFINITY) [reply src/frontend/src/lib/utils/convert.utils.ts:38]: Doch - the check stays because of the comments above :)

**Comment by DenysKarmazynDFINITY** [comment]
@AntonioVentilii ready for another look

**Review by AntonioVentilii — APPROVED** [review]
LGTM tks!

**Pull request merged**